### PR TITLE
Fixes core app conig check style error

### PR DESCRIPTION
**Pull request by mssfang**
Remove ThrowFromClientLoggerCheck from Implementation suppression list
Fixes for the ThrowFromClientLoggerCheck and MissingDeprecatedCheck
Fixes for ThrowFromClientLoggerCheck suppresion: Azure Core #5142 
Fixes for JavadocCodeSnippetCheck: AppConfiguration #5140

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 1]
```diff
@@ -157,7 +157,7 @@
      *
      * <p>Add a setting with the key "prodDBConnection" and value "db_connection".</p>
      *
-     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setSettingWithResponse#ConfigurationSetting}
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setSetting#string-string}
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: You changed this.. but did you also change the corresponding sample .java tag?

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/jackson/FlatteningSerializer.java [hunk 2]
```diff
@@ -41,6 +42,8 @@
  * {'properties' : { 'name' : 'my_name' }} in the serialized payload.
  */
 class FlatteningSerializer extends StdSerializer<Object> implements ResolvableSerializer {
+    private final transient ClientLogger logger = new ClientLogger(FlatteningSerializer.class);
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/jackson/FlatteningSerializer.java]: I'd remove the transient and suppress the Spotbug issue

Comment (conniey) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/jackson/FlatteningSerializer.java]: Why suppress the issue?

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java (hunk 1 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: I think it used a wrong codenippet sample. If you ctrl+shift+f to find "com.azure.data.appconfiguration.configurationasyncclient.setSetting#string-string" there is no one use this code sample. So I changed to current version.

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/jackson/FlatteningSerializer.java (hunk 2 above)
Comment (mssfang) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/jackson/FlatteningSerializer.java]: Yeah. The transient doesn't work here. Will suppress the issue. @conniey, local mavn run still pass but CI failed for this change

**Pull request merged**